public/app/features/alerting/unified/components/expressions/Expression.tsx

```diff
@@ -131,30 +132,82 @@ interface ExpressionResultProps {
   series: DataFrame[];
   isAlertCondition?: boolean;
 }
-
+export const PAGE_SIZE = 20;
 export const ExpressionResult: FC<ExpressionResultProps> = ({ series, isAlertCondition }) => {
+  const [pageIndex, setPageIndex] = useState(0);
   const styles = useStyles2(getStyles);
 
+  // reset the page index whenever the series changes
+  useEffect(() => {
+    setPageIndex(0);
+  }, [series, setPageIndex]);
+
   // sometimes we receive results where every value is just "null" when noData occurs
   const emptyResults = isEmptySeries(series);
   const isTimeSeriesResults = !emptyResults && isTimeSeriesFrames(series);
+  const resultPages = chunk(series, PAGE_SIZE);
```

Comment:
You can try to use `usePagination` hook (from (`public/app/features/alerting/unified/hooks/usePagination.ts`) to remove some of the pagination handling code.

Reply:
We thought that for this particular case it would be more simple and clear to use `chunk` from `lodash` (we only go forward and backward, we don't go directly to the page)

Reply:
Hmm, but you had to implement pagination logic once again (pageStart, pageEnd, clamp, chunk)

Reply:
Yea I think we can re-use that one, I forgot we had that :D It doesn't seem to expose the `start` and `end` of the current page but that's easy to add to the hook 👍

Reply:
It exposes `numberOfPages` which should be enough :)

Reply:
ahh ok! nice!